src/api/spec/models/workflow/step/branch_package_step_spec.rb

```diff
@@ -253,15 +253,16 @@
         end
       end
 
-      context 'for a push event' do
+      context 'with a push event for a commit' do
         let(:scm_webhook) do
           ScmWebhook.new(payload: {
                            scm: 'github',
                            event: 'push',
                            target_branch: 'main',
                            source_repository_full_name: 'reponame',
                            commit_sha: commit_sha,
-                           target_repository_full_name: 'openSUSE/open-build-service'
+                           target_repository_full_name: 'openSUSE/open-build-service',
+                           ref: 'refs/heads/branch_123'
```

Comment:
Adding the `ref` is needed only for GitHub since it's the only way to tell if a push event is for commits or tags. GitLab doesn't need this change since it has 2 separate push events for commits and tags.